apps/meteor/server/models/raw/LivechatDepartmentAgents.ts

```diff
@@ -395,6 +396,39 @@ export class LivechatDepartmentAgentsRaw extends BaseRaw<ILivechatDepartmentAgen
 	): FindCursor<ILivechatDepartmentAgents> {
 		return this.find({ agentId: { $in: agentsIds }, departmentId }, options);
 	}
+
+	findDepartmentsOfAgent(agentId: string, enabled = false): AggregationCursor<ILivechatDepartmentAgents & { departmentName: string }> {
+		return this.col.aggregate<ILivechatDepartmentAgents & { departmentName: string }>([
+			{
+				$match: {
+					agentId,
+					...(enabled && { departmentEnabled: true }),
+				},
+			},
+			{
+				$lookup: {
+					from: 'rocketchat_livechat_department',
+					localField: 'departmentId',
+					foreignField: '_id',
+					as: 'department',
+				},
+			},
+			{ $unwind: '$department' },
+			{
+				$project: {
+					_id: '$_id',
+					agentId: '$agentId',
+					departmentId: '$departmentId',
+					departmentName: '$department.name',
+					username: '$username',
+					count: '$count',
+					order: '$order',
+					_updatedAt: '$_udpatedAt',
```

Comment:
Not a big deal, but this specific property is not in the `ILivechatDepartmentAgents` type. Can you please add it?

Reply:
Actually this is not being used and it's just there because of the model typing. Gonna remove this from the return.
```suggestion
```

Reply:
Is it being returned, currently? If so, then removing it would be breaking :thinking:

Reply:
not breaking because it wasn't on the types 🤔 so even if it was there, there was no guarantee it should be there lol

Guess you can see it both ways

Reply:
But, just in case, you have it there now